discovery-provider/src/queries/get_trending_tracks.py

```diff
@@ -41,10 +43,14 @@ def generate_unpopulated_trending(
     time_range,
     strategy,
     exclude_premium=SHOULD_TRENDING_EXCLUDE_PREMIUM_TRACKS,
+    exclude_collectible_gated=SHOULD_TRENDING_EXCLUDE_COLLECTIBLE_GATED_TRACKS,
     limit=TRENDING_LIMIT,
 ):
+    # We use limit * 2 here to apply a soft limit so that
+    # when we later filter out premium or collectible gated tracks,
+    # we will probabilistically satisfy the given limit.
     trending_tracks = generate_trending(
-        session, time_range, genre, limit, 0, strategy.version
+        session, time_range, genre, limit * 2, 0, strategy.version
```

Comment:
will do a more thorough review in a bit but, any perf implications here?

Reply:
I current trending uses mat view so I don't think this code path will be hit in prod, but this old path is really bad performance wise